### DISCUSSION TITLE
Execute document writer destination mutiple times

### DISCUSSION THREAD

**Discussion by Parag411** [nextgenhealthcare/connect#5460] [General]
Hello guys,

I am converting XML files to HTML and then to PDF files.
if I place a for loop, still the destination is executed only once.
How can I convert multiple XML files by calling document writer of same destination again and again ?
I tried creating another channel and route the HTML code to that channel but it doesnt work.

Please let me know.

Thanks

**Comment by pacmano1** [comment]
Your second option where you interate over 1..n XML blobs and send them to a second channel via router.routemessage should work fine.   That is likely what most people do.  There are other ways of course.

**Reply by Parag411** [reply]
Could we set the same destination to execute repeatedly?

Thanks

**Reply by pacmano1** [reply]
You can read and adapt https://forums.mirthproject.io/forum/mirth-connect/support/13120-create-multiple-hl7-messages-out-of-raw-data-through-javascript for your use.     

That is sending mutliple messages at the same time to a destination.

**Reply by Parag411** [reply]
I went through the post, do I need to add below code in preprocessor to pass html code in document writer template ?
var messages = new java.util.ArrayList();
messages.add(message1);
messages.add(message2);
return messages;

**Reply by tonygermano** [reply]
A destination can only execute once per message. The code from that forum link is for generating multiple messages in a Javascript Reader. You can't follow that example if you are using a different source connector type to fetch/receive your messages.

The most common solution is what is mentioned first in this answer. In your for loop construct a new message with the data you will need for that pdf file, then route the message to a second channel (channel reader.) The second channel will have the destination with the document writer, which will be called once per message generated by your for loop. If you are only dealing with strings, you can pass all of the data using the sourceMap of the new message, it will be directly accessible by the document writer, and you will not need any transformers in the second channel to pull the data out and move it to other maps.
